bokehjs/src/lib/models/mappers/log_color_mapper.ts

```diff
@@ -26,53 +32,43 @@ export class LogColorMapper extends ContinuousColorMapper {
     this.prototype.type = "LogColorMapper"
   }
 
-  protected _v_compute<T>(data: Arrayable<number>, values: Arrayable<T>,
-      palette: Arrayable<T>, colors: {nan_color: T, low_color?: T, high_color?: T}): void {
-    const {nan_color, low_color, high_color} = colors
-
+  protected scan<T>(data: Arrayable<number>, palette: Arrayable<T>) : LogScanData {
     const n = palette.length
     const low = this.low != null ? this.low : min(data)
     const high = this.high != null ? this.high : max(data)
     const scale = n / (log1p(high) - log1p(low))  // subtract the low offset
+
+    return {high, low, scale}
+  }
+
+  protected cmap<T>(d : number,  palette: Arrayable<T>,  low_color: T,
+                    high_color: T, scan_data: LogScanData) : T {
+    // This handles the edge case where d == high, since the code below maps
+    // values exactly equal to high to palette.length, which is greater than
+    // max_key
     const max_key = palette.length - 1
 
-    for (let i = 0, end = data.length; i < end; i++) {
-      const d = data[i]
-
-      // Check NaN
-      if (isNaN(d)) {
-        values[i] = nan_color
-        continue
-      }
-
-      if (d > high) {
-        values[i] = high_color != null ? high_color : palette[max_key]
-        continue
-      }
-
-      // This handles the edge case where d == high, since the code below maps
-      // values exactly equal to high to palette.length, which is greater than
-      // max_key
-      if (d == high) {
-        values[i] = palette[max_key]
-        continue
-      }
-
-      if (d < low) {
-        values[i] = low_color != null ? low_color : palette[0]
-        continue
-      }
-
-      // Get the key
-      const log = log1p(d) - log1p(low)  // subtract the low offset
-      let key = Math.floor(log * scale)
-
-      // Deal with upper bound
-      if (key > max_key)
-        key = max_key
-
-      values[i] = palette[key]
+    if (d > scan_data.high) {
+      return high_color != null ? high_color : palette[max_key]
+    }
+    if (d == scan_data.high) {
```

Comment:
Not a big deal but I'd either keep the spacing between these ``if`` branches consistent or merge them into ``else if``

Reply:
Addressed in d2c9e2f5e